### PR TITLE
Vampire tail jab changes (try 2)

### PR DESCRIPTION
**Pull request by Red-byte3D**
# About the pull request
Makes tail jab only knock you down if you hit an obstacle.

# Explain why it's good for the game

A caste that already has healing, acid runner speed, very low cooldowns, dash, execute that heals even more and more does not need an auto aimed ability that stuns you just because you clicked near the marine.

Theres a difference between making the game accessible to newer players and completely removing any resemblance of skill / effort required to do well.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: Vampire tail jab now only stuns if you knockback an enemy into an obstacle.
spellcheck: Vampire lurkers no longer "throw it back" after hitting somebody with their Tail Jab
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm [hunk 1]
```diff
@@ -296,11 +306,7 @@
 	addtimer(CALLBACK(src, PROC_REF(reset_direction), xeno, last_dir, new_dir), 0.5 SECONDS)
 
 	hit_target.apply_armoured_damage(get_xeno_damage_slash(hit_target, xeno.caste.melee_damage_upper), ARMOR_MELEE, BRUTE, "chest")
-
-	if(hit_target.mob_size < MOB_SIZE_BIG)
-		hit_target.apply_effect(0.5, WEAKEN)
-	else
-		hit_target.apply_effect(0.5, SLOW)
+	hit_target.apply_effect(0.5, SLOW)
```
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:309]:
```suggestion
	hit_target.Slow(0.5)
```

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm [hunk 2]
```diff
@@ -282,9 +282,19 @@
...
+		return
+	if(!step(hit_target, direction))
+		playsound(hit_target.loc, "punch", 25, 1)
+		hit_target.visible_message(SPAN_DANGER("[hit_target] slams into an obstacle!"),
+		isxeno(hit_target) ? SPAN_XENODANGER("We slam into an obstacle!") : SPAN_HIGHDANGER("You slam into an obstacle!"), null, 4, CHAT_TYPE_TAKING_HIT)
+		hit_target.apply_damage(MELEE_FORCE_TIER_2)
+		if (hit_target.mob_size < MOB_SIZE_BIG)
+			hit_target.apply_effect(0.5, WEAKEN)
+		else
+			hit_target.apply_effects(0.5, SLOW)
```
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:297]:
```suggestion
			hit_target.Slow(0.5)
```

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm [hunk 3]
```diff
@@ -282,9 +282,19 @@
...
+	var/direction = Get_Compass_Dir(xeno, targeted_atom) //More precise than get_dir.
+	if(!direction) //Same tile.
+		return
+	if(!step(hit_target, direction))
+		playsound(hit_target.loc, "punch", 25, 1)
+		hit_target.visible_message(SPAN_DANGER("[hit_target] slams into an obstacle!"),
+		isxeno(hit_target) ? SPAN_XENODANGER("We slam into an obstacle!") : SPAN_HIGHDANGER("You slam into an obstacle!"), null, 4, CHAT_TYPE_TAKING_HIT)
+		hit_target.apply_damage(MELEE_FORCE_TIER_2)
+		if (hit_target.mob_size < MOB_SIZE_BIG)
+			hit_target.apply_effect(0.5, WEAKEN)
```
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:295]:
```suggestion
			hit_target.KnockDown(0.5)
```

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm [hunk 4]
```diff
@@ -282,9 +282,19 @@
...
 	stab_direction = turn(get_dir(xeno, targeted_atom), 180)
 	playsound(hit_target,'sound/weapons/alien_tail_attack.ogg', 50, TRUE)
-	if(hit_target.mob_size < MOB_SIZE_BIG)
-		step_away(hit_target, xeno)
 
+	var/direction = Get_Compass_Dir(xeno, targeted_atom) //More precise than get_dir.
+	if(!direction) //Same tile.
+		return
+	if(!step(hit_target, direction))
+		playsound(hit_target.loc, "punch", 25, 1)
```
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:290]:
```suggestion
		playsound(hit_target, "punch", 25, 1)
```

**Review by vero5123 — COMMENTED** [review]
Screen Shot 2024-06-04 at 12 46 23 PM

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm (hunk 4 above)
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:290]:
```suggestion
		playsound(hit_target, "punch", 25, 1)
```

**Comment by vero5123** [comment]
Balance aside, I feel like it would be a little more satisfying if smaller mobs actually got tossed onto the wall. 
Without flinging the mob 

https://github.com/cmss13-devs/cmss13/assets/73014819/e9435885-2d22-48fc-9830-2e8777e71350

throwing the mob

https://github.com/cmss13-devs/cmss13/assets/73014819/8206ae5b-4895-445a-8d40-e5f1e4331211

For reference I used the throw_carbon() proc.

**Comment by Diegoflores31** [comment]
this PR is basically removing Tail jab completely and making it a normal tail stab ??? having to rely on walls to do damage is a bad balance take .

**Comment by Drulikar** [comment]
> this PR is basically removing Tail jab completely and making it a normal tail stab ??? having to rely on walls to do damage is a bad balance take .

Of note, the ability still does its normal damage, its just a bit extra damage for slamming into a wall exactly like bullets that knockback mobs into an obstacle small some damage. The removal is the stun and making that only occur for wall slam.

**Comment by Diegoflores31** [comment]
> Of note, the ability still does its normal damage, its just a bit extra damage for slamming into a wall exactly like bullets that knockback mobs into an obstacle small some damage. The removal is the stun and making that only occur for wall slam.

This ability damage is the same damage than a a lurker tail stab and you cant target target it to limbs so the damage is very low even on direct hit. Considering it pushes you only 1 tile away . The situations where you somehow pull a stun is very scarce and the caste should not be relying on it to be able to kill a marine that can stun and kill you extremely quick.

File: code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm [hunk 5]
```diff
@@ -282,9 +282,19 @@
 
 	stab_direction = turn(get_dir(xeno, targeted_atom), 180)
 	playsound(hit_target,'sound/weapons/alien_tail_attack.ogg', 50, TRUE)
-	if(hit_target.mob_size < MOB_SIZE_BIG)
-		step_away(hit_target, xeno)
 
+	var/direction = Get_Compass_Dir(xeno, targeted_atom) //More precise than get_dir.
+	if(!direction) //Same tile.
+		return
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:288]: Why would we not perform damage if we're on the same tile? The only thing that should be skipped is the wallslam part of the code. We have yet to apply the slow, animation, damage, log, etc. yet.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
Also take into consideration that this will make execute harder to pull off too since that requires knock down or unconscious.

**Comment by Red-byte3D** [comment]
> Also take into consideration that this will make execute harder to pull off too since that requires knock down or unconscious.

does not matter in the contextg of execute, you can only really reliably execute somebody if they are crit or letting you

**Comment by Drulikar** [comment]
> > Also take into consideration that this will make execute harder to pull off too since that requires knock down or unconscious.
> 
> does not matter in the contextg of execute, you can only really reliably execute somebody if they are crit or letting you

I repeat, an execute can only be performed when the target is unconscious or knocked out. It *does* matter. The code isn't looking for "crit" or "they are letting you". 

https://github.com/cmss13-devs/cmss13/blob/d28283bae1c987f8e27e3e66d8459ee91a7b3e7f/code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm#L327-L329

**Review by Drulikar — APPROVED** [review]
Don't stress about what I was asking about; its simply that I was under the impression you could try to combo these together since I thought the KnockDown applied that trait that headbite is looking for.

But if it needs further tweaks I will consider making it part of a combo for example say you can only headbite as a follow up to jab or rush something, especially since there's now a visual indicator when they're a headbite eligible target - so now I could make more complicated requirements and still have that intuited by the player.

**Comment by Diegoflores31** [comment]
> But if it needs further tweaks I will consider making it part of a combo for example say you can only headbite as a follow up to jab or rush something, especially since there's now a visual indicator when they're a headbite eligible target - so now I could make more complicated requirements and still have that intuited by the player.

You cant headbite people unless they are inconcious.
Its not a part of the combo at all nor used to do damage , its just an ability to "glory kill" the marine when its already critted

**Comment by Drulikar** [comment]
It not *currently* part of a combo.

**Pull request merged**